lms/envs/docs/README.rst

```diff
@@ -0,0 +1,40 @@
+LMS Configuration Settings
+==========================
```

Comment:
👍 on documenting this!  Do you think there should also be a note regarding [derived settings](https://github.com/edx/edx-platform/blob/master/openedx/core/lib/derived.py) here?

Reply:
Yes, sure.  Although I'd prefer if someone from the Platform team did so.  I'm a bit confused on whether they are needed because of Django 1.11 upgrade technicalities or if it addresses a real need.  Also, I'm wondering whether those cases can be addressed via Django App Plugins instead.

Reply:
How about this:

Derived Settings
----------------

In cases where you need to define one or more settings relative to the value of another setting, you can explicitly designate them as derived calculations.  This can let you override one setting (such as a path or a feature toggle) and have it automatically propagate to other settings which are defined in terms of that value, without needing to track down all potentially impacted settings and explicitly override them as well.  This can be useful for test setting overrides even if you don't anticipate end users customizing the value very often.

For example:

```python
def _make_locale_paths(settings):
    locale_paths = [settings.REPO_ROOT + '/conf/locale']  # edx-platform/conf/locale/
    if settings.ENABLE_COMPREHENSIVE_THEMING:
        # Add locale paths to settings for comprehensive theming.
        for locale_path in settings.COMPREHENSIVE_THEME_LOCALE_PATHS:
            locale_paths += (path(locale_path), )
    return locale_paths
LOCALE_PATHS = _make_locale_paths
derived('LOCALE_PATHS')
```

In this case, `LOCALE_PATHS` will be defined correctly at the end of the settings module parsing no matter what `REPO_ROOT`, `ENABLE_COMPREHENSIVE_THEMING`, and `COMPREHENSIVE_THEME_LOCALE_PATHS` are currently set to.  This is true even if the `LOCALE_PATHS` calculation was defined in `lms/envs/common.py` and you're using `lms/envs/aws.py` which includes overrides both from that module and the JSON configuration files.

List entries and dictionary values can also be derived from other settings, even when nested within each other:

```python
def _make_mako_template_dirs(settings):
    """
    Derives the final Mako template directories list from other settings.
    """
    if settings.ENABLE_COMPREHENSIVE_THEMING:
        themes_dirs = get_theme_base_dirs_from_settings(settings.COMPREHENSIVE_THEME_DIRS)
        for theme in get_themes_unchecked(themes_dirs, settings.PROJECT_ROOT):
            if theme.themes_base_dir not in settings.MAKO_TEMPLATE_DIRS_BASE:
                settings.MAKO_TEMPLATE_DIRS_BASE.insert(0, theme.themes_base_dir)
    if settings.FEATURES.get('USE_MICROSITES', False) and getattr(settings, "MICROSITE_CONFIGURATION", False):
        settings.MAKO_TEMPLATE_DIRS_BASE.insert(0, settings.MICROSITE_ROOT_DIR)
    return settings.MAKO_TEMPLATE_DIRS_BASE

TEMPLATES = [
    {
        'NAME': 'django',
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        ...
    },
    {
        'NAME': 'mako',
        'BACKEND': 'edxmako.backend.Mako',
        'APP_DIRS': False,
        'DIRS': _make_mako_template_dirs,
    },
]
derived_collection_entry('TEMPLATES', 1, 'DIRS')
```

Reply:
thanks. updated.